code/API_definitions/EdgeCloud_LcM.yaml

```diff
@@ -0,0 +1,648 @@
+                        - VISIBILITY_INTERNAL
+                minItems: 1                
+              numOfInstances:
+                type: integer
+                format: int32
+                description: Number of component instances to be launched. 
+              restartPolicy:
+                type: string
+                enum:
+                  - RESTART_POLICY_ALWAYS
+                  - RESTART_POLICY_NEVER
+                description: How the platform shall handle component failure                        
+      required:
+        - name
+        - virtType
+        - fileName
+        - repository
+        - componentSpec
+    CpuArchType:
+          type: string
+          enum:
+            - x86_64
+            - arm_64
+          description: CPU Instruction Set Architecture (ISA) E.g., Intel, Arm etc. 
+    ProvisionedApp:
+      description: Information about the provisioned application
+      type: object
+      properties:
+        appInstanceId:
+          $ref: '#/components/schemas/AppInstanceId'
+        edgeResourceName:
+          $ref: '#/components/schemas/EdgeResourceName'
+    EdgeResourceName:
+      description: Edge Resource Name - an identifier for an edge reource in the operator domain 
+      type: string
+ 
+    ErrorInfo:
+      type: object
+      properties:
+        status:
+          type: integer
+          description: HTTP status code returned along with this error response
+        code:
+          type: string
+          description: Code given to this error
+        message:
+          type: string
+          description: Detailed error description
+      required:
+        - status
+        - code
+        - message
+    FileId:
+      type: string
+      format: uuid
+      description: A globally unique identifier assosiated with the image file. Originating OP generates this identifier when file is uploaded over NBI.
+ 
+    MecDetails:
+      type: object
+      required:
+        - geolocation
+      properties:
+        edgeResourceName: 
+          $ref: '#/components/schemas/EdgeResourceName'
+        status:
+          $ref: '#/components/schemas/Status'
+        region:
+            $ref: '#/components/schemas/RegionId'
+ 
+      minItems: 1
+ 
+    OperatingSystem:
+      type: object
+      required:
+        - architecture
+        - family
+        - version
+        - license
+      properties:
+        architecture: 
+          type: string
+          enum:
+            - x86_64
+            - x86
+          example: x86_64  
+        family:
+          type: string
+          enum:
+            - RHEL
+            - UBUNTU
+            - COREOS
+            - WINDOWS
+            - OTHERS
+        version:
+          type: string
+          enum:
+            - OS_VERSION_UBUNTU_2204_LTS
+            - OS_VERSION_RHEL_8
+            - OS_VERSION_RHEL_7
+            - OS_VERSION_ DEBIAN_11
+            - OS_VERSION_COREOS_STABLE
+            - OS_MS_WINDOWS_2012_R2
```

Comment:
Is this really needed? I guess we will end up with a very long list.

Reply:
I think if in the first version we only support containers type applications then we can possibly handle it better

Reply:
For now option `OTHER` has been added and some options have been kicked out to reduce the list